### PR TITLE
Add ARGs for cross compiling

### PR DESCRIPTION
**Pull request by sixeyed**
Bump Go to 1.8 and add build args. Defaults to Linux/x64 but can build for other platforms - e.g. 

```
docker build --build-arg GOOS=windows -t distribution-builder:windows .
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.7-alpine
+FROM golang:1.8-alpine
```
Comment (stevvooe) [inline Dockerfile]: I'm not sure we are quite on 1.8 yet.

**Comment by stevvooe** [comment]
Thanks for the fix!

We aren't quite on 1.8 yet, so if you could leave the Dockerfile on 1.7, we can get this merged.

**Comment by dmcgowan** [comment]
That update button doesn't work correctly, need to rebase locally and push up

**Comment by sixeyed** [comment]
By now we're on Alpine `1.8`, so just the `ARG` lines added.

**Comment by stevvooe** [comment]
@sixeyed Could you fix these to remove the merge commits?

**Comment by sixeyed** [comment]
My fork is badly diverged... Closing this and will open a new PR with a clean merge history.

**Pull request closed without merging**